### PR TITLE
Added PlaneHelper and Box3Helper

### PR DESCRIPTION
**Pull request by WestLangley**
As discussed in #11207.

/ping @Mugen87

There is not much here. We can fix it later if there are issues.

### CODE REVIEW

**Review by Mugen87 — APPROVED** [review]

**Comment by Mugen87** [comment]
Looks good 👍 . Just as a reminder: We should also add some docs in the future.

File: src/helpers/PlaneHelper.js [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+	this.type = 'PlaneHelper';
+
+	this.plane = plane;
+
+	this.size = size;
+
+	var color = ( hex !== undefined ) ? hex : 0xffff00;
+
+	if ( size === undefined ) size = 1;
```
Comment (mrdoob) [inline src/helpers/PlaneHelper.js]: Seems like size is not being used?

Comment (WestLangley) [inline src/helpers/PlaneHelper.js]: Oops. `this.size` is used in `update()`. I just have the code in the wrong order. Will fix.

Comment (WestLangley) [inline src/helpers/PlaneHelper.js]: Done. Thanks!

**Comment by mrdoob** [comment]
What do you guys think about this approach?

```js
import { Box3 } from '../math/Box3';
import { Line } from '../objects/Line';
import { LineBasicMaterial } from '../materials/LineBasicMaterial';
import { Float32BufferAttribute } from '../core/BufferAttribute';
import { BufferGeometry } from '../core/BufferGeometry';

/**
  * @author WestLangley / http://github.com/WestLangley
  *
 */

function PlaneHelper( plane, size, hex ) {

	this.type = 'PlaneHelper';

	if ( size === undefined ) size = 1;

	var color = ( hex !== undefined ) ? hex : 0xffff00;

	var positions = [ 1, - 1, 1, - 1, 1, 1, - 1, - 1, 1, 1, 1, 1, - 1, 1, 1, - 1, - 1, 1, 1, - 1, 1, 1, 1, 1, 0, 0, 1, 0, 0, 0 ];

	var geometry = new BufferGeometry();

	geometry.addAttribute( 'position', new Float32BufferAttribute( positions, 3 ) );

	Line.call( this, geometry, new LineBasicMaterial( { color: color } ) );

	this.geometry.computeBoundingSphere();

	this.onBeforeRender = function () {

		var scale = - plane.constant;

		if ( Math.abs( scale ) < 1e-8 ) scale = 1e-8; // sign does not matter

		this.scale.set( 0.5 * size, 0.5 * size, scale );

		this.lookAt( this.plane.normal );

	};

}

PlaneHelper.prototype = Object.create( Line.prototype );
PlaneHelper.prototype.constructor = PlaneHelper;

export { PlaneHelper };
```

Basically, making the helpers autonomous. The user no longer has to call `.update()` and they update at the right time. They will, of course, update redundantly most of the times, but maybe that's an ok price to pay for a much simpler API?

**Comment by Mugen87** [comment]
I wonder if it's possible to combine your approach with some kind of internal state mechanism. So automatic updates but only if one or more relevant values actually change. 

For example `PlaneHelper` could have additional variables like `_constant` and `_normal`. The helper compares these values to the actual plane properties. And if there are differences, the helper updates its internal state.

**Comment by WestLangley** [comment]
@mrdoob I like your suggestion. I see you are taking advantage of the closure, which is nice.

Perhaps we can just rename `update()` to `onBeforeRender()`, instead -- keeping it in the prototype.

If we merge this, I can file a new PR and experiment. We can revert if there are problems.

@Mugen87 I'm not sure performance is an issue for this helper because only uniforms are being updated.

**Comment by WestLangley** [comment]
OK, I removed `update()` in favor of `onBeforeRender`. Also rewrote `Box3Helper` so only uniforms are updated each frame.

**Comment by WestLangley** [comment]
@mrdoob Is this OK to merge? I have other PRs in the pipeline that depend on this.

**Comment by mrdoob** [comment]
Thanks!

**Comment by vidartf** [comment]
> Just as a reminder: We should also add some docs in the future.

Just wanted to point out that while this has now been released, the docs are still missing.

**Comment by looeee** [comment]
@vidartf could you add them?

**Comment by vidartf** [comment]
@looeee PR submitted as #12177

**Pull request merged**